packages/schemas/alterations/next-1684382842-add-name-tag-created-at-for-tenants-table.ts

```diff
@@ -0,0 +1,45 @@
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration.js';
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    // Create a trigger to set the tenant name since column reference is not available as default value.
+    await pool.query(sql`
+      create function set_tenant_name() returns trigger as
+      $$ begin
+        new.name := concat('tenant_', new.id);
+        return new;
+      end; $$ language plpgsql;
+    `);
+    await pool.query(sql`
+      create trigger set_tenant_name_trigger
+      before insert on tenants
+      for each row
+      execute procedure set_tenant_name();
+    `);
+    // Add new tenant columns for name, tag, and created_at.
+    await pool.query(sql`
```

Comment:
Should it run before defining the set_tenant_name function?

Reply:
Updated. I think the original order is also OK since the function/trigger will only be triggered by new rows and the alteration only operates on existing data rows.